_posts/2021-08-20-kingavis.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: post
+title: "[CVE-2023-3031] Improper Limitation of a Pathname to a Restricted Directory in Webbax - King-Avis module for PrestaShop"
+categories: module
+author:
+- Friends-Of-Presta.org
+meta: "CVE,PrestaShop,kingavis"
+severity: "medium (6.5)"
+---
+
+Improper Limitation of a Pathname leads to a Path Traversal vulnerability in the module King-Avis for Prestashop, allowing a user knowing the download token to read arbitrary local files.This issue affects King-Avis: before 17.3.15. 
+
+## Summary
+
+* **CVE ID**: [CVE-2023-3031](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-3031)
+* **Published at**: 2023-06-01 
+* **Advisory source**: [borelenzo.github.io](https://borelenzo.github.io/stuff/2023/06/01/cve-2023-3031.html)
+* **Vendor**: PrestaShop
+* **Product**: King-Avis
+* **Impacted release**: 17.3.15
+* **Product author**: Webbax
+* **Weakness**: [CWE-22](https://cwe.mitre.org/data/definitions/22.html)
+* **Severity**: medium (6.5)
+
+## Description
+
+[King-Avis](https://shop.webbax.ch/modules-utilitaire/161-module-prestashop-king-avis.html) is a Prestashop module developed by Webbax. In versions older than 17.3.15, the latter suffers from an authenticated path traversal, leading to local file read.
+
+There was a file download.php, that could be used to download statistical reports as CSV files. To protect from unauthorised access, the download feature was protected by a token.
+If the token is incorrect, the file exits and no content is returned. However, if the token is correct, the path is extracted from the parameter file and used without being sanitised.
+
+It means that administrators (supposed to know this token) can read arbitrary local files. Also, there is no need to have an active admin session to browse to this file.
+
+## CVSS base metrics
+
+* **Attack vector**: network
+* **Attack complexity**: low
+* **Privilege required**: low
+* **User interaction**: none
+* **Scope**: unchanged
+* **Confidentiality**: high
+* **Integrity**: none
+* **Availability**: none
+
+**Vector string**: [CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:N/A:N](https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?vector=AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:N/A:N)
+
+## Possible malicious usage
+
+* Stealing secrets to unlock admin controllers based on ajax script
+* Exfiltrate all modules with all versions to facilite pentesting
+* Stealing table_prefix to greatly facilitate SQL injections for kiddies who don’t know how exploit DBMS design’s vulnerabilities or steal database access to login in exposed PHPMyAdmin / Adminer / etc.
```

Comment:
who don’t know how exploit
...
, or stealing database access to log in to exposed PHPMyAdmin / Adminer / etc.